python/ray/serve/api.py

```diff
@@ -169,11 +169,11 @@ def shutdown(self) -> None:
             self._shutdown = True
 
     @_ensure_connected
-    def _get_result(self, result_object_id: ray.ObjectRef) -> bool:
-        result_id: UUID = ray.get(result_object_id)
-        result = ray.get(self._controller.wait_for_event.remote(result_id))
-        logger.debug(f"Getting result_id ({result_id}) with result: {result}")
-        return result
+    def _wait_for_goal(self, result_object_id: ray.ObjectRef) -> bool:
+        goal_id: Optional[UUID] = ray.get(result_object_id)
```

Comment:
made optional because some of the APIs return None when they're a no-op (for handling idempotence)